### ISSUE TITLE
docker 25.0.0 is no longer compatible with kind 0.20.0

### ISSUE THREAD

**Issue by alisterd51** [kubernetes-sigs/kind#3488]
**What happened**:

I built an image with the following command:
```bash
docker build . -t test
```
(dockerfile)
```Dockerfile
FROM ubuntu:latest
WORKDIR /app
CMD ["echo", "hello world"]
```
then I loaded it with the following command:
```bash
kind load docker-image test
```
and I get:
```log
Image: "test" with ID "sha256:e805d0fe428781e5c32394de2608c8b87ebc5957d5777eb2b0a7347d451406cc" not yet present on node "kind-control-plane", loading...
ERROR: failed to load image: command "docker exec --privileged -i kind-control-plane ctr --namespace=k8s.io images import --all-platforms --digests --snapshotter=overlayfs -" failed with error: exit status 1
Command Output: unpacking docker.io/library/test:latest (sha256:50807015a2a4824d0e7bc5f128ad0d78c71d1dea61aceeeef069fa745304948a)...time="2024-01-23T13:02:35Z" level=info msg="apply failure, attempting cleanup" error="wrong diff id calculated on extraction \"sha256:9ee40b9dc73697fc3c9f59e59d10d1d3391e58bdc35040b2a80469cf0273565c\"" key="extract-405237186-Reo_ sha256:8e87ff28f1b5ff2d5131999ccfa1e674cb252631c50683f5ee43fad59cbea8e1"
ctr: wrong diff id calculated on extraction "sha256:9ee40b9dc73697fc3c9f59e59d10d1d3391e58bdc35040b2a80469cf0273565c"
```

**What you expected to happen**:

I only expect to get something like:

```log
Image: "test" with ID "sha256:d74f45155991013b59487d7d1eec0e53fee88c1125c6256a5c07c8bbf8e889b7" not yet present on node "kind-control-plane", loading...
```
without any error message.

**How to reproduce it (as _minimally_ and precisely as possible)**:

detail before

**Anything else we need to know?**:

**Environment:**

- kind version: (use `kind version`):
```
kind v0.20.0 go1.21.1 linux/amd64
```
- Runtime info: (use `docker info` or `podman info`):
```
Client: Docker Engine - Community
 Version:    25.0.0
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 5
 Server Version: 25.0.0
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: a1496014c916f9e62104b33d1bb5bd03b0858e59
 runc version: v1.1.11-0-g4bccb38
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.5.0-14-generic
 Operating System: Ubuntu 23.10
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.34GiB
 Name: ***
 ID: ***
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: ***
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```
- OS (e.g. from `/etc/os-release`):
```
Ubuntu 23.10
```
- Kubernetes version: (use `kubectl version`):
```
Client Version: v1.28.6
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.27.3
```
- Any proxies or other special environment settings?:

**Comment by stmcginnis** [comment]
I believe there are various issues with Docker 25 at the moment. See #3487 for some details.

**Comment by PSanetra** [comment]
It seems like this issue will be resolved with Docker 25.0.1 (not yet released): https://github.com/moby/moby/pull/47161

**Comment by alisterd51** [comment]
fixed in docker [v25.0.1](https://github.com/moby/moby/releases/tag/v25.0.1)
